web/models/hn_story.ex

```diff
@@ -71,14 +80,17 @@ defmodule Feedya.HNStory do
     |> put_change(:hn_id, hn_id)
   end
 
-  defp fetch_many!(api_func) do
-    {:ok, ids} = api_func.()
-    ids_to_fetch = ids -- already_fetched(ids)
-    for id <- ids_to_fetch, do: spawn(
-      fn ->
-        id
-        |> get_story!
-        |> new!
-      end)
+  defp fetch_many!(get_ids) do
+    max_id = max_fetched
+    ids = get_ids.()
+    to_fetch = ids -- already_fetched(ids)
+
+    to_fetch
+    |> Flow.from_enumerable
+    |> Flow.partition(stages: 100)
+    |> Flow.map(&HN.story/1)
+    |> Flow.partition(stages: 100)
+    |> Flow.map(&create!/1)
```

Comment:
This is the place where race-conditions are likely to happen. 
But this is not a big problem, because we have unique index on `hn_id`, so we'll just end up with error in logs.